openedx/core/djangoapps/programs/utils.py

```diff
@@ -142,6 +179,11 @@ def progress(self, programs=None, count_only=True):
             list of dict, each containing information about a user's progress
                 towards completing a program.
         """
+        now = datetime.datetime.now(utc)
+        enrolled_run_modes = defaultdict(set)
+        for enrollment in self.enrollments:
+            enrolled_run_modes[str(enrollment.course_id)].add(enrollment.mode)
```

Comment:
i get the point about using a string for the value, but not sure how i would get the modes if i looped over course_run_ids, i wouldn't want to do a lookup every time?